spec/jobs/scheduled_publishing_job_spec.rb

```diff
@@ -10,12 +10,41 @@
       publish_service = double(:publish_service, publish: nil)
 
       expect(PublishService).to receive(:new).with(edition) { publish_service }
+      message_delivery = double(ActionMailer::MessageDelivery, deliver_later: nil)
 
       expect(publish_service)
         .to receive(:publish)
         .with(user: user, with_review: edition.status.details.reviewed)
 
-      ScheduledPublishingJob.new.perform(edition.id)
+      expect(ScheduledPublishMailer).to receive(:publish_email).with(edition, user)
+                                                               .and_return(message_delivery)
+
+      ScheduledPublishingJob.perform_now(edition.id)
+    end
+
+    it "calls ScheduledPublishMailer for each editor of the edition" do
+      editor_one = create(:user, email: "someone@example.com")
+      metadata_revision = create(:metadata_revision, scheduled_publishing_datetime: Time.current)
+      editor_two = create(:user, email: "someone-else@example.com")
+      revision = create(:revision,
+                        created_by: editor_two,
+                        metadata_revision: metadata_revision)
+      edition = create(:edition,
+                       :scheduled,
+                       revision: revision,
+                       created_by: editor_one)
+
+      publish_service = double(:publish_service, publish: nil)
+      message_delivery = double(ActionMailer::MessageDelivery, deliver_later: nil)
+
+      expect(PublishService).to receive(:new).with(edition) { publish_service }
+
+      edition.editors.each do |editor|
+        expect(ScheduledPublishMailer).to receive(:publish_email).with(edition, editor)
+                                                                 .and_return(message_delivery)
+      end
```

Comment:
Rather than using the edition.editors I think it'd be better to check each editor - as we're not actually testing here that each editor gets it. If editors returned an empty array we'd never run these expectations.

I'd set an`allow(ScheduledPublishMailer).to receive(:publish_email).and_return(message_delivery)` first

and then just have an expectation for each editor.

Reply:
Yep I agree, done!